### PR TITLE
Add os_log logging to Workers

### PR DESCRIPTION
**Pull request by dhavalshreyas**
Follow-up from #12, we had dropped the logging there, adding it back in here.

### CODE REVIEW

File: WorkflowReactiveSwift/Sources/Worker.swift [hunk 1]
```diff
@@ -63,14 +63,33 @@ struct WorkerWorkflow<WorkerType: Worker>: Workflow {
     }
 
     func render(state: State, context: RenderContext<WorkerWorkflow>) -> Rendering {
+        let logger = WorkerLogger<WorkerType>()
+
         // Start with Void to ensure `worker.run()` is called only once for a given key
         SignalProducer(value: ())
-            .flatMap(.latest) { self.worker.run() }
+            .flatMap(.latest) {
+                logger.logStarted()
```
[the remark this answers is not in the document]
Comment (bencochran) [reply WorkflowReactiveSwift/Sources/Worker.swift:71]: Like maybe an `on(started:)` hanging off the `.run()`?

File: WorkflowReactiveSwift/Sources/Logger.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    // MARK: - Workers
+
+    func logStarted() {
+        if #available(iOS 12.0, macOS 10.14, *) {
+            os_signpost(
+                .begin,
+                log: .worker,
+                name: "Running",
+                signpostID: self.signpostID,
+                "Worker: %{public}@",
```
Comment (bencochran) [inline WorkflowReactiveSwift/Sources/Logger.swift:41]: Should we consider these non-public? If I were implementing a Worker to do login, I’d likely put a username & password into the properties of that Worker.

Comment (AquaGeek) [reply WorkflowReactiveSwift/Sources/Logger.swift:41]: We're logging the type of the worker, not the worker itself.

Realistically, we can probably move to making all of these non-public, though.

Comment (dhavalshreyas) [reply WorkflowReactiveSwift/Sources/Logger.swift:41]: Cool, will go ahead and make these private?

Comment (dhavalshreyas) [reply WorkflowReactiveSwift/Sources/Logger.swift]: Done.

Comment (bencochran) [reply WorkflowReactiveSwift/Sources/Logger.swift]: Oops, totally missed that it was the type not the value.

**Pull request merged**